### PR TITLE
DOC/BLD: update README.md, remove setuptools from dependency

### PR DESCRIPTION
**Pull request by fangchenli**
xref #41815

### CODE REVIEW

**Comment by jreback** [comment]
https://github.com/pandas-dev/pandas/blob/master/setup.cfg#L33 update as well i think

File: README.md [hunk 1]
```diff
@@ -100,9 +101,10 @@ pip install pandas
 ```
 
 ## Dependencies
+- [Setuptools - A fully-featured, actively-maintained, and stable library designed to facilitate packaging Python projects](https://setuptools.readthedocs.io/en/latest/)
```
Comment (jorisvandenbossche) [inline README.md:104]: Can you move this to the optional dependencies section?

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
I wouldn't add it to `install_requires`, see https://github.com/pandas-dev/pandas/issues/41815

File: doc/source/getting_started/install.rst [hunk 2]
```diff
@@ -221,7 +221,6 @@ Dependencies
 ================================================================ ==========================
 Package                                                          Minimum supported version
 ================================================================ ==========================
-`setuptools <https://setuptools.readthedocs.io/en/latest/>`__    38.6.0
```
Comment (jorisvandenbossche) [inline doc/source/getting_started/install.rst:224]: Can you add it to the optional dependencies below?

**Review by simonjayhawkins — APPROVED** [review]

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**